-1

Tested at: 66b0f42714738ce02778dfde32fbfc48a152a69a

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23444/28418/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/plugins/SurveyDBUploader.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/plugins/SurveyInputTrackerFromDB.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/src/SurveyAlignmentSensor.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/src/SurveyParameters.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.cc:42:6: error: prototype for 'void SurveyInputCSCfromPins::orient(LocalVector, LocalVector, double, double, double&, double&, double&, double&, double&, double&)' does not match any in class 'SurveyInputCSCfromPins'
 void SurveyInputCSCfromPins::orient(LocalVector LC1, LocalVector LC2, double a, double b, double &T, double &dx, double &dy, double &dz, double &PhX, double &PhZ) {
      ^~~~~~~~~~~~~~~~~~~~~~
In file included from /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.cc:21:0:
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.h:28:8: error: candidate is: void SurveyInputCSCfromPins::orient(align::LocalVector, align::LocalVector, double, double, double&, double&, double&, double&, double&, double&)
   void orient(align::LocalVector LC1, align::LocalVector LC2, double a, double b, double &T, double &dx, double &dy, double &dz, double &PhX, double &PhZ);
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/test/SurveyInputDummy.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/test/SurveyTest.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/test/SurveyInputTest.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/test/SurveyDBReader.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/test/Modules.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.cc:42:30: error: out-of-line definition of 'orient' does not match any declaration in 'SurveyInputCSCfromPins'
void SurveyInputCSCfromPins::orient(LocalVector LC1, LocalVector LC2, double a, double b, double &T, double &dx, double &dy, double &dz, double &PhX, double &PhZ) {
                             ^~~~~~
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.h:28:15: note: type of 1st parameter of member declaration does not match definition ('Vector3DBase<align::Scalar, [...]>' vs 'Vector3DBase<float, [...]>')
  void orient(align::LocalVector LC1, align::LocalVector LC2, double a, double b, double &T, double &dx, double &dy, double &dz, double &PhX, double &PhZ);
              ^
</pre>

--- a/Alignment/LaserAlignment/plugins/RawDataConverter.h
+++ b/Alignment/LaserAlignment/plugins/RawDataConverter.h
@@ -2,6 +2,8 @@
 #include "Alignment/LaserAlignment/interface/LASGlobalData.h"
 #include <DataFormats/Common/interface/DetSetVector.h>
 #include <FWCore/Framework/interface/Event.h> 
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "Alignment/LaserAlignment/interface/LASGlobalLoop.h"
 
 // Forward declarations 
 class TFile;

--- a/Alignment/SurveyAnalysis/plugins/SurveyDataConverter.h
+++ b/Alignment/SurveyAnalysis/plugins/SurveyDataConverter.h
@@ -21,6 +21,7 @@
 #include "Alignment/SurveyAnalysis/interface/SurveyDataReader.h"
 #include "FWCore/Framework/interface/EDAnalyzer.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "Alignment/TrackerAlignment/interface/TrackerAlignment.h"
 
 class SurveyDataConverter : public edm::EDAnalyzer
 {

--- a/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.cc
+++ b/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.cc
@@ -39,7 +39,7 @@ SurveyInputCSCfromPins::SurveyInputCSCfromPins(const edm::ParameterSet& cfg)
   , m_stationErrorPhiZ(cfg.getParameter<double>("stationErrorPhiZ"))
 {}
 
-void SurveyInputCSCfromPins::orient(LocalVector LC1, LocalVector LC2, double a, double b, double &T, double &dx, double &dy, double &dz, double &PhX, double &PhZ) {
+void SurveyInputCSCfromPins::orient(align::LocalVector LC1, align::LocalVector LC2, double a, double b, double &T, double &dx, double &dy, double &dz, double &PhX, double &PhZ) {
    double cosPhX, sinPhX, cosPhZ, sinPhZ;
 
    LocalPoint LP1(LC1.x(), LC1.y() + a, LC1.z() + b);

--- a/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.h
+++ b/Alignment/SurveyAnalysis/plugins/SurveyInputCSCfromPins.h
@@ -11,6 +11,7 @@
  */
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "Alignment/SurveyAnalysis/interface/SurveyInputBase.h"
+#include "CondFormats/Alignment/interface/Definitions.h"
 
 class SurveyInputCSCfromPins:
   public SurveyInputBase
@@ -24,7 +25,7 @@ public:
 
 private:
 
-  void orient(LocalVector LC1, LocalVector LC2, double a, double b, double &T, double &dx, double &dy, double &dz, double &PhX, double &PhZ);
+  void orient(align::LocalVector LC1, align::LocalVector LC2, double a, double b, double &T, double &dx, double &dy, double &dz, double &PhX, double &PhZ);
   void errors(double a, double b, bool missing1, bool missing2, double &dx_dx, double &dy_dy, double &dz_dz, double &phix_phix, double &phiz_phiz, double &dy_phix);
 
   void fillAllRecords(Alignable *ali);

--- a/Alignment/TrackerAlignment/plugins/TrackerSystematicMisalignments.h
+++ b/Alignment/TrackerAlignment/plugins/TrackerSystematicMisalignments.h
@@ -13,7 +13,8 @@
 
 #include "FWCore/Framework/interface/ESHandle.h"
 #include "FWCore/Framework/interface/EDAnalyzer.h"
-
+#include "Alignment/CommonAlignment/interface/Alignable.h"
+#include "Alignment/TrackerAlignment/interface/AlignableTracker.h"
 
 class AlignableSurface;
 class Alignments;

